packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -133,274 +125,33 @@ export default class VTable {
      * @param format Table format to apply
      */
     applyFormat(format: Partial<TableFormat>) {
-        if (!this.table) {
+        if (!this.table || !format) {
             return;
         }
-
-        this.formatInfo = format ? format : this.formatInfo;
-        this.table.style.borderCollapse = 'collapse';
-        this.setBordersType(this.formatInfo);
-        this.setColor(this.formatInfo);
-        this.setFirstColumnFormat(this.formatInfo);
-        this.setHeaderRowFormat(this.formatInfo);
+        this.deleteCellShadeDataset(this.cells);
+        applyTableFormat(this.table, this.cells, format);

```

Comment:
you can call writeBack() to trigger the applyTableFormat() call. See the design chart.